lib/chipmunk/incoming_storage.rb

```diff
@@ -1,12 +1,53 @@
 # frozen_string_literal: true
 
+require "pathname"
+
 module Chipmunk
-  # IncomingStorage is a factory for proxies to storage of incoming deposits.
+  # IncomingStorage is responsible for the business rules of writing to and reading from
+  # Volumes earmarked for its use. It is specifically concerned with initial upload of bags,
+  # and makes them available to other parts of the system via its methods.
   class IncomingStorage
 
+    class UserPathBuilder
```

Comment:
I would say that these are not strictly builders, and not really factories. The naming here is a little hard, though, since we do want to capture the root and localize the supplied paths within it...

Maybe these are `UploadPath`, defaulting to the ID segment only, with a specialization strategy of `UploadPath::UserOwned`?

With that pattern, `for` as the method reads better to me.

Reply:
I'd probably pull these up, out of the IncomingStorage namespace, too, just to make them easier to find/reference.